discovery.rst

```diff
@@ -123,53 +147,89 @@ This type of discovery finds a URI factory for a PSR-7_ URI implementation::
 Integrating your own implementation with the discovery mechanism
 ----------------------------------------------------------------
 
-The ``php-http/discovery`` has built-in support for some implementations.
-To use a different implementation or override the default when several implementations are available in your code base,
-you can register a class explicitly with the corresponding discovery service. For example::
-
-    HttpClientDiscovery::register('Acme\MyClient', true);
-
-- ``class``: The class that is instantiated. This class MUST NOT require any constructor arguments.
-- ``condition``: The condition that is evaluated to boolean to decide whether the class can be instantiated.
+In order to make your class discoverable, you have to configure it as a Puli resource
```

Comment:
I would have one phrase first that says this is when writing your own Httplug client or factory, just to be more clear when this is relevant.